backend/src/game/engine.py

```diff
@@ -0,0 +1,43 @@
+from backend.src.game.status import Status
+
+
+class GameEngine():
+
+    def __init__(self):
+        self.players = []
+        self.alive_players = []
+        self.dead_players = []
+        self.game_status = Status.GAME_OVER
+
+    def add_player(self, player):
+        self.players.append(player)
+
+    def begin_game(self):
+        self.alive_players = self.players
+        self.game_status = Status.ACTIVE
+
+    def play_game(self):
+        while self.game_status:
+            for player in self.yield_next_player():
+                self.take_turn(player)
+                self.validate_if_winner(player)
+                if not self.game_status:
+                    break
+                self.set_dead_players()
+
+    def take_turn(self, player):
+        pass
+
+    def validate_if_winner(self, player):
+        if player.victory_points == 20:
+            self.game_status = Status.GAME_OVER
+
+    def set_dead_players(self):
```

Comment:
How do you see this method working? When is this triggered, how is it used?
The way I had it in my head:
At the end of every turn, the dice are handled. Damage is dealt. At this point, players would be marked dead if conditions are met. In which case I don't think we need this method. All players start as alive and then when they're marked dead, they're moved.
But this doesn't mean that's the way it has to be. Looking forward to what's in your head.